libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -711,6 +718,194 @@ void AP_DDS_Client::on_request(uxrSession* uxr_session, uxrObjectId object_id, u
         GCS_SEND_TEXT(MAV_SEVERITY_INFO, "%s Request for Mode Switch : %s", msg_prefix, mode_switch_response.status ? "SUCCESS" : "FAIL");
         break;
     }
+    case services[to_underlying(ServiceIndex::SET_PARAMETERS)].rep_id: {
+        rcl_interfaces_srv_SetParameters_Request set_parameter_request;
+        rcl_interfaces_srv_SetParameters_Response set_parameter_response;
+        const bool deserialize_success = rcl_interfaces_srv_SetParameters_Request_deserialize_topic(ub, &set_parameter_request);
+        if (deserialize_success == false) {
+            break;
+        }
+
+        if (set_parameter_request.parameters_size > 8U) {
+            break;
+        }
+
+        // Set parameters and responses for each one requested
+        set_parameter_response.results_size = set_parameter_request.parameters_size;
+        for (size_t i = 0; i < set_parameter_request.parameters_size; i++) {
+            rcl_interfaces_msg_Parameter param = set_parameter_request.parameters[i];
+
+            enum ap_var_type var_type;
+
+            // set parameter
+            AP_Param *vp;
+            char param_key[AP_MAX_NAME_SIZE+1];
```

Comment:
```suggestion
            char param_key[AP_MAX_NAME_SIZE + 1];
```

Style nit - `astyle` doesn't catch these.